uniffi_macros/src/export/trait_interface.rs

```diff
@@ -30,15 +29,32 @@ pub(super) fn gen_trait_scaffolding(
     let trait_impl = callback_interface::trait_impl(mod_path, &self_ident, &items)
         .unwrap_or_else(|e| e.into_compile_error());
 
+    let clone_fn_ident = Ident::new(
+        &uniffi_meta::clone_fn_symbol_name(mod_path, &trait_name),
+        Span::call_site(),
+    );
     let free_fn_ident = Ident::new(
-        &free_fn_symbol_name(mod_path, &trait_name),
+        &uniffi_meta::free_fn_symbol_name(mod_path, &trait_name),
         Span::call_site(),
     );
 
-    let free_tokens = quote! {
+    let helper_fn_tokens = quote! {
+        #[doc(hidden)]
+        #[no_mangle]
+        pub unsafe extern "C" fn #clone_fn_ident(
+            ptr: *const ::std::ffi::c_void,
+            call_status: &mut ::uniffi::RustCallStatus
+        ) -> *const ::std::ffi::c_void {
+            uniffi::rust_call(call_status, || {
+                let ptr = ptr as *mut std::sync::Arc<dyn #self_ident>;
+                let arc = unsafe { ::std::sync::Arc::clone(&*ptr) };
+                Ok(::std::boxed::Box::into_raw(::std::boxed::Box::new(arc)) as  *const ::std::ffi::c_void)
+            })
+        }
+
         #[doc(hidden)]
         #[no_mangle]
-        pub extern "C" fn #free_fn_ident(
+        pub unsafe extern "C" fn #free_fn_ident(
```

Comment:
Clippy complained about the lack of unsafe in the clone function.  I think free is just as unsafe, but clippy doesn't recognize it, so I added the keyword here to.  It shouldn't make a difference, these functions are only called across the FFI anyways.

Reply:
please add a one-liner of that as a comment :)